lightning-invoice/src/utils.rs

```diff
@@ -630,7 +641,7 @@ fn sort_and_filter_channels<L: Deref>(
 				// look at the public channels instead.
 				log_trace!(logger, "Not including channels in invoice route hints on account of public channel {}",
 					log_bytes!(channel.channel_id));
-				return vec![]
+				return vec![].into_iter().take(3).map(route_hint_from_channel);
```

Comment:
Do we need the `take(3)` here and below? We already do this within `_create_phantom_invoice` and if removed here should also do it in `_create_invoice_from_channelmanager_and_duration_since_epoch_with_payment_hash` with a common constant. Otherwise, it's hidden that we are taking only 3.

Reply:
From what I understand we need the `take(3)` in `sort_and_filter_channels`.  We could move out to the file scope `const MAX_HINTS: usize = 3;` for reuse in the functions that build up to `_create_phantom_invoice`.  We could rename it to `MAX_CHANNEL_HINTS` for clarity.


My understanding of the need for `take(3)`:
The reason for the `take(3)` in `sort_and_filter_channels` is to have a `ExactSizeIterator` as the return type and the reason for that is the check on the returned length in `select_phantom_hints`.
https://github.com/lightningdevkit/rust-lightning/blob/3e6b9aed12b304d8bb1f7756353dc11bdeb286fe/lightning-invoice/src/utils.rs#L250

Reply:
Ah, I see. Yeah, could you make a file-level constant for all uses?